src/main/java/org/spongepowered/api/item/data/EnchantmentData.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.item.data;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.item.Enchantment;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Represents item data related to an item having active {@link Enchantment}s.
+ *
+ * <p>Some enchantments are not compatible with some item types, so checking
+ * on the enchantment before setting is recommended.</p>
+ */
+public interface EnchantmentData extends ItemData<EnchantmentData> {
+
+    /**
+     * Gets all the enchantments attached to the item stack.
+     *
+     * @return All enchantments attached to the item stack
+     */
+    Set<Enchantment> getEnchantments();
+
+    /**
+     * Gets all the enchantments and the levels attached to the item stack.
+     *
+     * @return A map containing all enchantments and their levels
+     */
+    Map<Enchantment, Integer> getEnchantmentLevels();
+
+    /**
+     * Gets the enchantment level of the desired {@link Enchantment} on the
+     * item stack.
+     *
+     * @param enchantment The enchantment
+     * @return The level of the enchantment, if available
+     */
+    Optional<Integer> getEnchantmentLevel(Enchantment enchantment);
+
+    /**
+     * Sets the enchantment with the desired level.
+     *
+     * <p>It is checked that the level meets the minimum and maximum
```

Comment:
Does this throw an IllegalArgumentException like Bukkit did or is it truncated automatically?